src/video/emscripten/SDL_emscriptenevents.c

```diff
@@ -488,7 +498,7 @@ Emscripten_HandleKeyPress(int eventType, const EmscriptenKeyboardEvent *keyEvent
     char text[5];
     Emscripten_ConvertUTF32toUTF8(keyEvent->charCode, text);
     SDL_SendKeyboardText(text);
-    return 1;
+    return SDL_GetEventState(SDL_TEXTINPUT) == SDL_ENABLE;
```

Comment:
I tested this and navigation on the page with the default event states does not work, exactly like before and like desired.